graph.py

```diff
@@ -36,19 +36,19 @@ def add_node(self, n):
         """
         if self.has_node(n):
             raise KeyError('Node already in graph.')
-        self[n] = set()
+        self[n] = dict()
 
-    def add_edge(self, n1, n2):
+    def add_edge(self, node1, node2, edge_weight):
         """Add a new edge connecting n1 to n2. Will implicitly create
         n1 and n2 if either do not exist.
         """
-        if not self.has_node(n2):
-            self.add_node(n2)
+        if not self.has_node(node2):
+            self.add_node(node2)
         try:
-            self[n1].add(n2)
+            self[node1].update({node2: edge_weight})
         except KeyError:
-            self.add_node(n1)
-            self[n1].add(n2)
+            self.add_node(node1)
+            self[node1].update({node2: edge_weight})
```

Comment:
For consistency, I think it'd be best to use `node` throughout your code, or `n`. You switch around in this script. Descriptive multi-letter variables are best, so I like what you did here.